gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergRegisterStepTest.java

```diff
@@ -100,4 +140,17 @@ protected IcebergCatalog mockSingleTableIcebergCatalog(IcebergTable mockTable) t
     Mockito.when(catalog.openTable(any())).thenReturn(mockTable);
     return catalog;
   }
+
+  protected IcebergTable mockIcebergTableForRetryer(String tableUuid, String tableMetadataFileLocation, Boolean throwOnlyExceptionsForRegisterIcebergTable) throws IOException {
+    TableMetadata mockMetadata = mockTableMetadata(tableUuid, tableMetadataFileLocation);
+
+    IcebergTable mockTable = Mockito.mock(IcebergTable.class);
+    Mockito.when(mockTable.accessTableMetadata()).thenReturn(mockMetadata);
+    if (throwOnlyExceptionsForRegisterIcebergTable) {
+      Mockito.doThrow(new RuntimeException()).when(mockTable).registerIcebergTable(any(), any());
+    } else {
+      Mockito.doThrow(new RuntimeException()).doThrow(new RuntimeException()).doNothing().when(mockTable).registerIcebergTable(any(), any());
+    }
```

Comment:
Would recommend doing this in respective test methods instead of doing it here which gives a better understanding to the reader.

Reply:
Removed the method.